### ISSUE TITLE
Bring back the old interface

### ISSUE THREAD

**Issue by lawyer88** [Alex313031/Thorium-Win#248]
I updated the browser to the latest version and found that the interface was disfigured by the latest one in Chrome...
For me personally, when switching to Torium, the main advantage was the ability to use a minimalistic interface... How can I get it back? Bleeding from the eyes from the “New Design”... I wouldn’t want to look for a new suitable browser again

**Comment by Alex313031** [comment]
@lawyer88 Already ahead of you.

Im working on a hybrid UI. It keeps the tab search button on the right (it already is changed that way now), but also restores the tab height, tab separators, and the square shape of menus.
Some things like icons and colors are left in the chrome 2023 UI style.

See commit https://github.com/Alex313031/thorium/commit/07bd646449bedc683b833e2c931747ad4aee27ed

Im calling it "Thorium 2024 UI" or "Th23", to mimick the Cr23 that it aims to fix

**Comment by lawyer88** [comment]
> @lawyer88 Already ahead of you.
> 
> Im working on a hybrid UI. It keeps the tab search button on the right (it already is changed that way now), but also restores the tab height, tab separators, and the square shape of menus. Some things like icons and colors are left in the chrome 2023 UI style.
> 
> See commit [Alex313031/thorium@07bd646](https://github.com/Alex313031/thorium/commit/07bd646449bedc683b833e2c931747ad4aee27ed)
> 
> Im calling it "Thorium 2024 UI" or "Th23", to mimick the Cr23 that it aims to fix

Thank you very much for your answer, but after clicking on the link I didn’t understand anything, unfortunately :) because I don't have any special knowledge in this area.
Could you help return the old browser interface to the way it was on the previous version before the update? What needs to be done for this?

**Comment by joshbali** [comment]
@Alex313031
The new 2023 Refresh UI is such an eyesore. One of the reasons I switched from vanilla Chrome to Thorium was that you kept the old UI, including the icons. I'm glad to hear there's hope that we'll get at least some of that back. I have been patiently waiting for the new 126 release, but was a bit shocked to read that the 2023 Refresh UI is now the Thorium default. Thank you for all your work.

**Comment by tigers-aye** [comment]
Thanks so much for working on this @Alex313031, and for the Thorium fork itself.  The atrocious 2023 UI is what brought me to Thorium in the first place, and I must say, it's fantastic work you've done.

Glad to know the 'old UI' will be reintegrated as well. I plan to remain on Version 124.0.6367.218 until it's ready (that's how much I despise the 2023 UI!). 

>  Some things like icons and colors are left in the chrome 2023 UI style.

Small request - is it possible to retain the yellow bookmark folder icon? 

I don't know why Chromium (and Firefox) opted to replace it with the colorless 'wireframe' folder icon. It just looks so drab in comparison, like a folder skeleton..

**Comment by lawyer88** [comment]
> I plan to remain on Version 124.0.6367.218 until it's ready (that's how much I despise the 2023 UI!).

Thanks for the advice! I uninstalled the new version and installed the old one) I had no complaints about it))

**Comment by SolenoidMan** [comment]
I too migrated to Thorium to avoid the terrible 126 UI but running 124 is simply not an option due to some of the major security holes that were patched after that version. Running 124 to avoid the UI is very risky business since so many sites are now abusing one of the flaws. But at least Thorium plays nicer with the terrible UI than all the other chromium forks I've tried. Thorium is still the best bet though it's a shame all of the chromium browsers are now visually ruined for the time being.

**Comment by tigers-aye** [comment]
@SolenoidMan - thanks for heads up on the v124 security issues. Just installed the upgrade. Figured no sense in taking a security risk over minor/temporary UI gripes.

**Comment by Aedon** [comment]
For me, the [#force-enable-tab-outlines](chrome://flags/#force-enable-tab-outlines) came in clutch. Earlier it was useless but after cr23 this is essential for me (I use a dark theme).

**Comment by Markosz22** [comment]
> Im working on a hybrid UI. It keeps the tab search button on the right (it already is changed that way now), but also restores the tab height, tab separators, and the square shape of menus.

That sounds great, one of the reasons I left Chrome for Firefox was the forced UI changes. I hate the overly rounded buttons and large empty space. Would it be possible to make the height of UI elements configurable, like height of tab, search and favourites separability?
![image](https://github.com/user-attachments/assets/432eae7f-7d74-4c67-b72c-ca1f1891ad98)

Chrome takes so much more space away from the screen

**Comment by SolenoidMan** [comment]
> For me, the #force-enable-tab-outlines came in clutch. Earlier it was useless but after cr23 this is essential for me (I use a dark theme).

Thankyou very much for this reply, I use black themes and this was a major improvement in visibility.

**Comment by Templayer** [comment]
I'm still on Thorium 122 and this is how I set up my Thorium: :) (make it fullscreen to see the trapezoid details!)

![image](https://github.com/user-attachments/assets/1d084064-9d4e-49df-b69b-27a4a817ba63)

**Comment by Delphox** [comment]
Honestly my main issue with the new interface is that hitting the top left corner of the screen no longer opens the first tab due to the padding they added. Kinda messed with years of muscle memory lol.

**Comment by Alex313031** [comment]
@Delphox @Templayer @joshbali @Markosz22 @lawyer88 @tigers-aye @Malus-risus New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246

**Comment by Templayer** [comment]
> @Delphox @Templayer @joshbali @Markosz22 @lawyer88 @tigers-aye @Malus-risus New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246

I have to admit, this is kinda awesome.

Haven't tested it (I'm tired of everything being unpolished nowadays), but can't wait to get a stable and polished version for this. I always wanted somebody to middle-finger them like this.

**Comment by Malus-risus** [comment]
> > @lawyer88 Already ahead of you.
> > Im working on a hybrid UI. It keeps the tab search button on the right (it already is changed that way now), but also restores the tab height, tab separators, and the square shape of menus. Some things like icons and colors are left in the chrome 2023 UI style.
> > See commit [Alex313031/thorium@07bd646](https://github.com/Alex313031/thorium/commit/07bd646449bedc683b833e2c931747ad4aee27ed)
> > Im calling it "Thorium 2024 UI" or "Th23", to mimick the Cr23 that it aims to fix
> 
> I like the UI design of Fluent Design. Could you incorporate some elements of that design into the interface, even if it's just partial?

@Alex313031  Do you have this plan?

**Comment by lawyer88** [comment]
> > @Delphox @Templayer @joshbali @Markosz22 @lawyer88 @tigers-aye @Malus-risus New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246
> 
> I have to admit, this is kinda awesome.
> 
> Haven't tested it (I'm tired of everything being unpolished nowadays), but can't wait to get a stable and polished version for this. I always wanted somebody to middle-finger them like this.

It's great, but how can you apply all this to the torium so that everything returns to the old design?) I didn’t see the instructions in the description (or didn’t understand it)... :) 
PS. Since the design change, he left for Vivaldi... There are minor shortcomings in working with the browser, but the alternative to the "broken (I hope temporarily) torium" is worthy.

Please ask - is it possible to write down detailed instructions for returning the old design? Without special knowledge it is difficult to understand :(

**Comment by Templayer** [comment]
> > > @Delphox @Templayer @joshbali @Markosz22 @lawyer88 @tigers-aye @Malus-risus New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246
> > 
> > 
> > I have to admit, this is kinda awesome.
> > Haven't tested it (I'm tired of everything being unpolished nowadays), but can't wait to get a stable and polished version for this. I always wanted somebody to middle-finger them like this.
> 
> It's great, but how can you apply all this to the torium so that everything returns to the old design?) I didn’t see the instructions in the description (or didn’t understand it)... :) PS. Since the design change, i am left for Vivaldi... There are minor shortcomings in working with the browser, but the alternative to the "broken (I hope temporarily) torium" is worthy.
> 
> Please ask - is it possible to write down detailed instructions for returning the old design? Without special knowledge it is difficult to understand :(

It is written in the post you have quoted.

> You will have to go to chrome://flags and enable the new chrome://flags/#thorium-2024 flag.

So in the URL bar (where you USUALLY enter the URL addresses of websites), enter chrome://flags/#thorium-2024 and enable it.

And, of course, you have to download and install (or use the portable) the test Thorium version from the link you have quoted. If you do not know how to install a browser, then use www.google.com

**Comment by Delphox** [comment]
> @Delphox @Templayer @joshbali @Markosz22 @lawyer88 @tigers-aye @Malus-risus New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246

I tested it on a live Linux environment and like it visually! It did seem to create an invisible border at the top when not using the system titlebar which would make switching tabs very annoying (as well as not fixing the corner click issue I mentioned earlier), but I'd assume this is part of the bugs you mentioned earlier and will be addressed.

https://github.com/user-attachments/assets/32be5d1b-8133-4b6c-89c5-c4991bd2fd50

Here's a quick recording of what I mentioned (first is 124 with the ideal behavior, then 126 with the top click working but not the corner click, and then enabling the flag where neither works).

**Comment by lawyer88** [comment]
> So in the URL bar (where you USUALLY enter the URL addresses of websites), enter chrome://flags/#thorium-2024 and enable it.
> 
> And, of course, you have to download and install (or use the portable) the test Thorium version from the link you have quoted

Thank you very much! 
I searched for this flag in the installed version, not in the portable version. I don’t use portable, it doesn’t suit me: (I’ll wait for the official release with these changes:)

**Comment by Templayer** [comment]
> > So in the URL bar (where you USUALLY enter the URL addresses of websites), enter chrome://flags/#thorium-2024 and enable it.
> > And, of course, you have to download and install (or use the portable) the test Thorium version from the link you have quoted
> 
> Thank you very much! I searched for this flag in the installed version, not in the portable version. I don’t use portable, it doesn’t suit me: (I’ll wait for the official release with these changes:)

Yup, I'm waiting for a polished and spotless version myself. :P

**Comment by trlkly** [comment]
I'm glad you're working on it!

I want to clarify some problems with the menus. Sure, the corners and whitespace make it take up way too much room. But it seems the biggest culprit of needing to scroll the menu is that the menu always spawns at the around the same horizontal coordinate of the mouse cursor. And it always cuts off way too early, well before it reaches the task bar. 

This is not how standard menus work. When you right click, the menu will appear at the same Y-coordinate **if all the items will fit.** It will keep expanding until the bottom right coordinate reaches the task bar  But if the items don't fit,  the top left coordinate moves up. 

Only once the top coordinate reaches up to the top of the screen (and the bottom coordinate is just above the taskbar) will the menu start to scroll.

Microsoft Edge implements the overly large curved menus. However, it does at least handle the menu behavior correctly. 

Note, if the menu is small enough, it may spawn the bottom left coordinate at the mouse cursor instead of the top right. This is desirable behavior, but irrelevant to the scrolling issue.

**Comment by trlkly** [comment]
> @Delphox @Templayer @joshbali @Markosz22 @lawyer88 @tigers-aye @Malus-risus New beta "Th24" release that fixes this and more > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.246

Any chance you can release a Windows build for testing as well?

**Comment by joshbali** [comment]
@Alex313031
Tested the Windows beta in a Windows 11 VM. Very promising so far. I still miss the old orange folder icons in the bookmarks, but the rest is much better. I'm looking forward to the first stable release for Windows. Thanks Alex for all your work. Much appreciated.

**Comment by Alex313031** [comment]
@Delphox The tab selection issue is addressed in the new Beta 2 release > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.251

@joshbali I will add that to my todo, to restore the old icons when Th24 is enabled.

**Comment by firegold763** [comment]
> @Delphox The tab selection issue is addressed in the new Beta 2 release > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.251
> 
> @joshbali I will add that to my todo, to restore the old icons when Th24 is enabled.
> 
> NOTE: Merging issue [Alex313031/thorium#769](https://github.com/Alex313031/thorium/issues/769) into this one.

Thank you for your work, I tried the second beta on Windows and the theme is already very nice. Can't wait for the more stable version on M127!

**Comment by trlkly** [comment]
> Number of bookmarks you can open before it prompts "Are you sure you want to open XXX tabs?" was upped from 15 to 50. (Personal pet peeve of mine)

Any chance this can be a tweakable setting?

I don't think I've ever intentionally middle-clicked on a bookmark folder. I like that it warns me when I do that.

**Comment by joshbali** [comment]
@Alex313031
After testing the Windows Th24 beta a few times in a Win11 VM, I finally decided to try it as my everyday browser to see how it works in real life. No major issues so far and for what I do with and in my browser (web application development + normal use) and with my settings and extensions, everything works as expected. Nice, love it. One step closer to what I was hoping for. Thanks Alex.

**Comment by trlkly** [comment]
@firegold763 How did you test the second beta on Windows? I can't find a built yet. Did you build it yourself, or is it available elsewhere?

**Comment by Delphox** [comment]
> @Delphox The tab selection issue is addressed in the new Beta 2 release > https://github.com/Alex313031/thorium/releases/tag/M126.0.6478.251
> 
> @joshbali I will add that to my todo, to restore the old icons when Th24 is enabled.
> 
> NOTE: Merging issue [Alex313031/thorium#769](https://github.com/Alex313031/thorium/issues/769) into this one.

Thanks! I can confirm it's fixed and its looking very good. If possible I'd like you to look into making the first tab clickable when clicking on the top-left corner/side as it was in v124, but if you don't wanna remove that padding for aesthetic reasons I understand, this is still a lot better.

**Comment by Alex313031** [comment]
@Delphox This isn't really feasible without a bunch more redesigning of the tab strip. Probably wont be adding that, sorry. However, I might make tabs closer to the left to make the first one easier to click.

**Comment by Templayer** [comment]
> @Delphox This isn't really feasible without a bunch more redesigning of the tab strip. Probably wont be adding that, sorry. However, I might make tabs closer to the left to make the first one easier to click.

It may be apt to give us an option to disable/enable that. :p

**Comment by Alex313031** [comment]
@trlkly @Templayer @Delphox @firegold763 New Windows Beta 2 just released, that has these fixes and more, as well as a new flag to set a custom tab width.

**Comment by trlkly** [comment]
Unfortunately, the color picker in the theme customizer seems to have been lost. IT used to let you pick any color on screen, but now only lets you pick between a bunch of very light colors. 

Thing is, my old theme color did work--until I tried customizing it. Then I couldn't set it back. So the code is still there to set it to whatever color you want. It just seems to be inaccessible now. Even if I had to type in a color code, I would like it back.

Though, if you reimplement using the OS them color (which is an option on Chrome proper), that would work for my use case. That's all I used it for: to make the title bar the same color as every other window.

**Comment by joshbali** [comment]
@Alex313031
Fantastic job. Love the new Beta2 Th24 build. Apart from the ugly monochrome bookmark icons ( no worries, I know... it's on your list), there's only one thing I'd suggest for now, and that's adding another value for the default tab width. It would be nice if you could add 180 pixels, as the jump from 120 to 240 is quite significant, and for me (personal opinion only) it's a jump from too small to too large. Other than that, the new build looks perfect to me. No massive UI changes, but none of the annoying and stupid changes of the vanilla Chrome 2023 Refresh UI, plus the forced MV2 extensions and some other subtle but quite clever changes. Looking forward to the next release.

**Comment by Alex313031** [comment]
@joshbali OK, I will add 180px, and bookmarks icon is in the works. Will probably stop there, and finalize it for the stable M128 release. I might add even more widths (I just don't want to add a textbox, if you read the release notes, and I also don't want the dropdown menu for the sizes to get to long)

Future work on Th24 will not be done after the above until the next release after M128.

**Comment by Templayer** [comment]
> @joshbali OK, I will add 180px, and bookmarks icon is in the works. Will probably stop there, and finalize it for the stable M128 release. I might add even more widths (I just don't want to add a textbox, if you read the release notes, and I also don't want the dropdown menu for the sizes to get to long)
> 
> Future work on Th24 will not be done after the above until the next release after M128.

Sounds like a plan. This way, even people like me who require stable releases can enjoy it. :)

**Comment by gz83** [comment]
If you have any further questions, please feel free to let us know.
